### PR TITLE
Fix #16470 - [Android] Improvements to TabbedPage API

### LINKED ISSUE #16470: Maui migrating  Xamarin to Maui - Tabbed Page Scroll Issue - Tabs are not scrolling
### Description

Hi, we are migrating from Xamarin to Maui, so we want to implement this with minimal changes otherwise it will be a big change and we don't want to use ShellTabs.

-Minimum target API is 27

-Expected behavior is tabs should scroll Below is Tabbed Page code

Tab

### Steps to Reproduce

1. Open the Maui application
2. Run and see the behavior of tabs
3. It should be scrollable 
4. Added Tabbar.xml file in Resources/Layout/Tabbar.xml. Added below code in project file also,
<ItemGroup><AndroidResource Include="Resources\Layout\Tabbar.axml" /></ItemGroup>

### Link to public reproduction project repository

https://github.com/siddhisadh/MauiAppTabs

### Version with bug

Unknown/Other

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 9 and Up

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by widavies**
### Description of Change

Both changes modify the API of `TabbedPage`

1) Added `BarTabMode` which can be set on `TabbedPage` to make tabs fixed or scrollable
2) Added `BarIndicatorColor` to set the underline color for a `TabbedPage`

### Issues Fixed

Fixes #16470

### CODE REVIEW

File: src/Controls/src/Core/PublicAPI/net-windows/PublicAPI.Unshipped.txt [hunk 1]
```diff
@@ -34,7 +34,12 @@ Microsoft.Maui.Controls.PointerGestureRecognizer.PointerMovedCommandParameter.ge
 *REMOVED*Microsoft.Maui.Controls.PointerGestureRecognizer.PointerExitedCommandParameter.get -> System.Windows.Input.ICommand!
 *REMOVED*Microsoft.Maui.Controls.PointerGestureRecognizer.PointerMovedCommandParameter.get -> System.Windows.Input.ICommand!
 Microsoft.Maui.Controls.Handlers.Items.StructuredItemsViewHandler<TItemsView>.~StructuredItemsViewHandler() -> void
+<<<<<<< Updated upstream
 Microsoft.Maui.Controls.ShellNavigationQueryParameters.TryGetValue(string! key, out object! value) -> bool
+=======
+Microsoft.Maui.Controls.TabbedPage.BarTabMode.get -> Microsoft.Maui.BarTabMode
+Microsoft.Maui.Controls.TabbedPage.BarTabMode.set -> void
+>>>>>>> Stashed changes
```
Comment (jfversluis) [inline src/Controls/src/Core/PublicAPI/net-windows/PublicAPI.Unshipped.txt:42]: This doesn't look right

File: src/Controls/src/Core/PublicAPI/net-windows/PublicAPI.Unshipped.txt [hunk 2]
```diff
@@ -81,7 +86,11 @@ override Microsoft.Maui.Controls.SearchBar.IsEnabledCore.get -> bool
 ~override Microsoft.Maui.Controls.Shapes.Matrix.Equals(object obj) -> bool
 ~static Microsoft.Maui.Controls.Handlers.ShellHandler.MapFlyoutIcon(Microsoft.Maui.Controls.Handlers.ShellHandler handler, Microsoft.Maui.Controls.Shell view) -> void
 ~static Microsoft.Maui.Controls.Region.FromRectangles(System.Collections.Generic.IEnumerable<Microsoft.Maui.Graphics.Rect> rectangles) -> Microsoft.Maui.Controls.Region
+<<<<<<< Updated upstream
 ~static readonly Microsoft.Maui.Controls.InputView.IsTextPredictionEnabledProperty -> Microsoft.Maui.Controls.BindableProperty
+=======
+~static readonly Microsoft.Maui.Controls.TabbedPage.BarTabModeProperty -> Microsoft.Maui.Controls.BindableProperty
+>>>>>>> Stashed changes
```
Comment (jfversluis) [inline src/Controls/src/Core/PublicAPI/net-windows/PublicAPI.Unshipped.txt:93]: This doesn't look right

**Review by jfversluis — CHANGES_REQUESTED** [review]
Couple of git merge indicators in there.

Additionally, I'm not sure if this is the way to implement this. This is very specific to Android and would confuse people trying to use this on other platforms. While this might fix the issue at hand, I feel that we should have a more platform specific way of doing this.

**Comment by widavies** [comment]
> Couple of git merge indicators in there.
> 
> Additionally, I'm not sure if this is the way to implement this. This is very specific to Android and would confuse people trying to use this on other platforms. While this might fix the issue at hand, I feel that we should have a more platform specific way of doing this.

Good point, I had that thought when I was working on this - I would assume iOS has identical flags on their tab view, but I'm not certain. Maybe for now I could add the feature as an Android specific flag, similar to this example for toolbar placement from your [official docs](https://learn.microsoft.com/en-us/dotnet/maui/android/platform-specifics/tabbedpage-toolbar-placement):
```xaml
<TabbedPage ...
            xmlns:android="clr-namespace:Microsoft.Maui.Controls.PlatformConfiguration.AndroidSpecific;assembly=Microsoft.Maui.Controls"
            android:TabbedPage.ToolbarPlacement="Bottom">
    ...
</TabbedPage>
```

Oops, looks like a bad merge on the API there. I'll go ahead and fix that. Just to confirm - I'm assuming that I'll add the new API to the "unshipped" API listing, and then if this PR gets approves it will get moved to the "shipped" API listing? Or should I put it on the "shipped" API listing to begin with?

Thanks

**Comment by PureWeen** [comment]
Apologies @widavies, but can we look at adding this one into the Maui Community Toolkit as an Attached Property?

https://github.com/CommunityToolkit/Maui

Unfortunately, we're past the point of being able to take new APIs right now for .NET 8. The other option would be to discuss this one for .NET 9 in a few months.

**Comment by widavies** [comment]
@PureWeen No worries, sure, I'll look into creating a pull request in the community toolkit. Do we want to close this PR or should we keep it open for .NET 9?

**Comment by samhouts** [comment]
Probably best to close it for now. After GA, we can reopen it or create a new one. Thanks!!

**Pull request closed without merging**